source/ota.c

```diff
@@ -1336,8 +1337,19 @@ static OtaErr_t shutdownHandler( const OtaEventData_t * pEventData )
     /* If we're here, we're shutting down the OTA agent. Free up all resources and quit. */
     agentShutdownCleanup();
 
-    /* Clear the entire agent context. This includes the OTA agent state. */
-    ( void ) memset( &otaAgent, 0, sizeof( otaAgent ) );
+    /* Clear the entire agent context except for pOtaInterface and OtaAppCallback
+     * to avoid accessing NULL function pointer if other task is running. Don't
+     * reset isOtaInterfaceInited to ensure os.event won't be created again in the
+     * next OTA_Init. */
+    ( void ) memset( &otaAgent.pThingName, 0, sizeof( otaAgent.pThingName ) );
```

Comment:
Initializing the pointers and other fields to zero value would be better in terms of readability here.
```
otaAgent.pThingName = NULL;
otaAgent.fileContext = NULL;
....
otaAgent.statistics = 0;
```

Reply:
Fixed in the latest commit.